src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -0,0 +1,584 @@
+                                        mergeFeedsResult.remappedIds.put(key, val);
+                                    }
+                                }
+                                // If a service_id from the active calendar has both the start_date and end_date in the
+                                // future, the service will be excluded from the merged file. Records in trips,
+                                // calendar_dates, and calendar_attributes referencing this service_id shall also be
+                                // removed/ignored. Stop_time records for the ignored trips shall also be removed.
+                                if (feedIndex > 0) {
+                                    int startDateIndex = getFieldIndex(fieldsFoundInZip, "start_date");
+                                    LocalDate startDate = LocalDate.parse(csvReader.get(startDateIndex), GTFS_DATE_FORMATTER);
+                                    if (startDate.isAfter(LocalDate.now())) {
+                                        LOG.warn("Skipping calendar entry {} because it operates in the future.", keyValue);
+                                        String key = String.join(":", table.name, idScope, keyValue);
+                                        mergeFeedsResult.skippedIds.add(key);
+                                        skipRecord = true;
+                                        continue;
+                                    }
+                                    // If a service_id from the active calendar has only the end_date in the future, the
+                                    // end_date shall be set to one day prior to the earliest start_date in future
+                                    // dataset before appending the calendar record to the merged file.
+                                    int endDateIndex = getFieldIndex(fieldsFoundInZip, "end_date");
+                                    if (index == endDateIndex) {
+                                        LocalDate endDate = LocalDate.parse(csvReader.get(endDateIndex), GTFS_DATE_FORMATTER);
+                                        if (endDate.isAfter(LocalDate.now())) {
+                                            val = feedsToMerge.get(0).version.validationResult.firstCalendarDate
+                                                .minus(1, ChronoUnit.DAYS)
+                                                .format(GTFS_DATE_FORMATTER);
+                                        }
+                                    }
+                                }
+                                break;
+                            case "trips":
+                                // trip_ids between active and future datasets must not match. If any trip_id is found
+                                // to be matching, the merge should fail with appropriate notification to user with the
+                                // cause of the failure. Merge result should include all conflicting trip_ids.
```

Comment:
Seems like the trip_id could just be changed to another UUID to avoid conflicts altogether. Maybe that's worth creating an issue for perhaps.

Reply:
Failing the merge on trip_id conflict is an MTC requirement, maybe I can make that a little more clear in the comment.

Reply:
I think this might be ok after I realized what is happening down at https://github.com/conveyal/datatools-server/pull/186/files#r272345429. If so, mark this as resolved and let's discuss at that other comment.